samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/CinemaApiModel.java

```diff
@@ -0,0 +1,65 @@
+package com.example.cinema.model;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.List;
+
+public interface CinemaApiModel {
+    sealed interface ShowCommand {
+
+      record CreateShow(String title, int maxSeats) implements ShowCommand {}
+
+      record ReserveSeat(String walletId, String reservationId, int seatNumber) implements ShowCommand {}
+
+      record ConfirmReservationPayment(String reservationId) implements ShowCommand {}
+
+          record CancelSeatReservation(String reservationId) implements ShowCommand {}
+    }
+
+    enum ShowCommandError {
+      SHOW_ALREADY_EXISTS,
+      SHOW_NOT_FOUND,
+      TOO_MANY_SEATS,
+      SEAT_NOT_FOUND,
+      SEAT_NOT_AVAILABLE,
+      RESERVATION_NOT_FOUND,
+      DUPLICATED_COMMAND,
+      CANCELLING_CONFIRMED_RESERVATION
+    }
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
+    @JsonSubTypes({
+      @JsonSubTypes.Type(value = Response.Success.class),
+      @JsonSubTypes.Type(value = Response.Failure.class)
+    })
+    sealed interface Response {
```

Comment:
This can probably move to `com.example.common` package. 

We had a similar situation in the user registry choreography sample and we move it to `common` package.

Reply:
agreed